mkosi/config.py

```diff
@@ -3293,12 +3315,16 @@ def resolve_deps(images: Sequence[argparse.Namespace], include: Sequence[str]) -
 
 
 def parse_config(argv: Sequence[str] = (), *, resources: Path = Path("/")) -> tuple[Args, tuple[Config, ...]]:
-    # Compare inodes instead of paths so we can't get tricked by bind mounts and such.
-    namespace = argparse.Namespace()
-    defaults = argparse.Namespace()
-    parsed_includes: set[tuple[int, int]] = set()
-    immutable_settings: set[str] = set()
-    append = False
+
+    class State:
```

Comment:
Renamed to `ParseContext`

We can't make it a dataclass that we instantiate anew every time because the inner functions would not get the new object but still operate on the old one